### PR TITLE
Adds Particle Network (Wallet-as-a-Service)

### PR DESCRIPTION
**Pull request by TABASCOatw**
### Description
This PR adds Particle Network as a supported wallet.

Particle Network is a Wallet-as-a-Service provider leveraging social logins to facilitate interaction via embedded wallet.

Beyond providing the capability of displaying Particle as a standalone wallet provider, functionality to display Particle as a purely social login shortcut has also been included, adding further flexibility to `web3-onboard`.

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [ ] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [ ] Add/update the appropriate package README (if applicable)
- [ ] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [ ] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

**Comment by leightkt** [comment]
@TABASCOatw thank you for submitting! I don't currently see any changes to the demo package, which is were all the testing should be conducted. Will you please add this new module to the demo and test and commit those changes? 
If you follow this PR example, you can see the changes to the demo: [PR EXAMPLE](https://github.com/blocknative/web3-onboard/pull/1238/files#diff-ae81ee2e749e1966947f7955cc7b3f6f3be7424ca3622674f2928062fcc1f775)

**Comment by TABASCOatw** [comment]
> @TABASCOatw thank you for submitting! I don't currently see any changes to the demo package, which is were all the testing should be conducted. Will you please add this new module to the demo and test and commit those changes? If you follow this PR example, you can see the changes to the demo: [PR EXAMPLE](https://github.com/blocknative/web3-onboard/pull/1238/files#diff-ae81ee2e749e1966947f7955cc7b3f6f3be7424ca3622674f2928062fcc1f775)

Sure, np, how's this?

File: packages/particle-network/package.json
Comment (leightkt) [inline packages/particle-network/package.json:3]:
```suggestion
  "version": "2.0.0-alpha-1",
```

File: packages/demo/package.json
Comment (leightkt) [inline packages/demo/package.json:65]:
```suggestion
    "@web3-onboard/particle-network": "^2.0.0-alpha.1",
```

**Comment by leightkt** [comment]
@TABASCOatw I'm seeing a bunch of errors on start up of the demo- can you take a look?

**Comment by TABASCOatw** [comment]
Sure, will check

**Comment by TABASCOatw** [comment]
> @TABASCOatw I'm seeing a bunch of errors on start up of the demo- can you take a look?

Are you testing it locally? The changes to the demo in this PR assumes `@web3-onboard/particle-network` exists, which it wont until this is merged- this could potentially be the issue? (if so, would need to import locally when testing) Could you share some of the specific errors that you're seeing?

**Comment by TABASCOatw** [comment]
> > @TABASCOatw I'm seeing a bunch of errors on start up of the demo- can you take a look?
> 
> Are you testing it locally? The changes to the demo in this PR assumes `@web3-onboard/particle-network` exists, which it wont until this is merged- this could potentially be the issue? (if so, would need to import locally when testing) Could you share some of the specific errors that you're seeing?

Bumping this quickly @leightkt

**Comment by leightkt** [comment]
@TABASCOatw while running the demo package (which pulls packages from the repo so as far as it is concerned @web3-onboard/particle-network exists. You need to run the demo and run through the checklist for tests with demo app above. They are checked off currently, but if you haven't run the package in the demo app- those tests are incomplete

**Comment by leightkt** [comment]
@TABASCOatw demo now builds- icon for particle is missing, and when I try to connect my wallet I get an error: 
Screenshot 2023-12-21 at 10 26 01 AM
Screenshot 2023-12-21 at 10 26 18 AM

**Comment by TABASCOatw** [comment]
> @TABASCOatw demo now builds- icon for particle is missing, and when I try to connect my wallet I get an error: Screenshot 2023-12-21 at 10 26 01 AM Screenshot 2023-12-21 at 10 26 18 AM

Apologies, I just updated it, there was a small error in the initial update with the way in which the image was referenced. Should be good to go now, tested everything locally and all test functions work properly. Let me know if it's working on your end, and thank you for the continued help here!

**Comment by TABASCOatw** [comment]
Let me know if it works now - happy to make any additional changes if needed aswell!

**Comment by leightkt** [comment]
@TABASCOatw this is looking good to me- was able to run through all the tests successfully in the demo! 
tagginb @Adamj1232 to take a look

**Review by leightkt — APPROVED** [review]

File: packages/particle-network/src/index.ts
Comment (Adamj1232) [inline packages/particle-network/src/index.ts:10]: We try to dynamically import any dependencies (that are type imports) to keep weight off the client until selected by a user. Can we update these to be imported dynamically?

File: packages/particle-network/package.json
Comment (Adamj1232) [inline packages/particle-network/package.json:62]: Are there any build environment configs that would need to be called out in the README and docs for these deps?
Example of the supported build env: https://onboard.blocknative.com/docs/modules/core#build-environments

**Review by Adamj1232 — CHANGES_REQUESTED** [review]
@TABASCOatw looking good and thank you for your contribution to the codebase! 🔥 

One question around build envs for the module and a request to dynamically import the deps within the index file

File: packages/particle-network/src/index.ts
Comment (Adamj1232) [inline packages/particle-network/src/index.ts:34]: Can you please run the prettier formatter when you have a moment on this file? This will allow the particle module to match the rest of the codebase in layout and allow for easier IDE readability 🤓

File: packages/particle-network/src/index.ts
Comment (Adamj1232) [inline packages/particle-network/src/index.ts:30]: Not a requirement but adding JSDocs comments to these types allows for better devEx and something we strive for

File: packages/particle-network/src/index.ts
Comment (TABASCOatw) [reply packages/particle-network/src/index.ts:45]: Sure, just ran it through the formatter, should be good to go now

File: packages/particle-network/package.json
Comment (TABASCOatw) [reply packages/particle-network/package.json:62]: Not with these SDKs - our newer SDKs do have a buffer requirement but these two will work out-of-the-box.

File: packages/particle-network/src/index.ts
Comment (TABASCOatw) [reply packages/particle-network/src/index.ts]: I can do this with a few objects, such as `createEIP1193Provider`, `ParticleNetwork`, and `ParticleProvider` but run into issues when trying to convert various type imports; I've shifted the import structure here to be similar to the Web3Auth package, let me know if this works.

**Comment by TABASCOatw** [comment]
> @TABASCOatw looking good and thank you for your contribution to the codebase! 🔥
> 
> One question around build envs for the module and a request to dynamically import the deps within the index file

Hey Adam, thank you! I've just updated everything, should be good to go now, let me know if this works 👍

File: packages/particle-network/src/index.ts
Comment (Adamj1232) [inline packages/particle-network/src/index.ts:60]: Can we move these into the `getInterface` function so they are only imported once a use selects particle?

Comment (TABASCOatw) [reply packages/particle-network/src/index.ts]: Sure! Just shifted it

**Review by Adamj1232 — APPROVED** [review]
Awesome! Thank you for collaborating with us! 🚀

**Comment by leightkt** [comment]
@TABASCOatw if you run `yarn && yarn check-all` on the repo- the particle-network package is throwing a few type errors- can you run and take a look to see if you can resolve? thanks!

**Comment by TABASCOatw** [comment]
> @TABASCOatw if you run `yarn && yarn check-all` on the repo- the particle-network package is throwing a few type errors- can you run and take a look to see if you can resolve? thanks!

Sure np! Just resolved it, passes all checks now 👍

**Comment by leightkt** [comment]
@TABASCOatw thanks for resolving those type errors! I went in and tried to run the demo to just make sure everything is still working correctly and it won't start with particle as an option for onboard wallets- can you take a look? thanks!

**Comment by TABASCOatw** [comment]
> @TABASCOatw thanks for resolving those type errors! I went in and tried to run the demo to just make sure everything is still working correctly and it won't start with particle as an option for onboard wallets- can you take a look? thanks!

Hey @leightkt! Apologies for the delay here, just fixed it, should work now 👍

**Comment by TABASCOatw** [comment]
GM @leightkt! Have you had a chance to give it a try? Happy to make any other modifications if need be

**Comment by TABASCOatw** [comment]
Quick bump on the above

**Comment by leightkt** [comment]
@TABASCOatw apologies for the delay in my response- I was out of office. Looking and testing good!

**Pull request merged**
